src/shared/hostname-setup.c

```diff
@@ -164,6 +183,14 @@ int hostname_setup(bool really) {
                 }
         }
 
+        if (!hn) {
+                r = acquire_hostname_from_credential(&b);
+                if (r >= 0) {
+                        hn = b;
+                        source = HOSTNAME_TRANSIENT;
+                }
+        }
```

Comment:
hmm, please move this up, before reading of /etc/hostname (i.e. after reading the kernel command line). We generally want that transient hostnames are overriden by explicitly configured ones.

Reply:
configured should trump transient.

(this used to be the other way round a long time ago, but was switched eventually. many years ago though already.)

Reply:
this matches behaviour of configuraiton files, where /etc/ always wins over /run/ too in our codebase. i.e. local admin always gets the final say.

Reply:
> configured should trump transient

Isn't that exactly how this works, then?

Reply:
@poettering ping -- I'm not sure what the issue is here, the code I wrote considers the credential after `/etc/hostname`.